### PR TITLE
Add In_windows_eventlog2

### PR DESCRIPTION
**Pull request by cosmo0920**
This plugin uses `EvtXXX` API series.

### Tasks

- [x] Depends on winevt_c gem which provides API from wevtapi.h
- [x] Subscribe channels with new API
- [x] Support tailing and resuming operations 
- [x] Create parser plugin for parsing winevt's XML log
  - [x] Create its plugin test cases
- [x] Use parser plugin helper in in_windows_eventlog2 plugin
- [x] Handle message from DLLs
- [x] Import KEY_MAP feature from in_windows_eventlog
- [x] Write documentation for in_windows_eventlog2

---

Related to https://github.com/fluent/fluent-plugin-windows-eventlog/pull/19.

### CODE REVIEW

**Comment by cosmo0920** [comment]
@repeatedly This PR is ready to review.
Could you kindly take a look?

**Comment by cosmo0920** [comment]
This PR is ready to review.

I've tested `in_windows_eventlog2` for `Application`, `Security`, `System`, and `Setup` channels.
`in_windows_eventlog2` does not cause SEGV and works normally.

Kernel-General, Microsoft-Windows-Servicing, and some sources' descriptions are not correctly handled but win32-eventlog gem also cannot handle them.
In this PR, I decided to align behavior against in_windows_eventlog (and its dependency of win32-eventlog gem).

Could you kindly take a look, @repeatedly ?

File: lib/fluent/plugin/parser_winevt_xml.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+require 'nokogiri'
+
+module Fluent::Plugin
+  class WinevtXMLparser < Parser
+    Fluent::Plugin.register_parser('winevt_xml', self)
+
+    def parse(text)
+      record = {}
+      doc = Nokogiri::XML(text)
+      record["ProviderName"]          = (doc/'Event'/'System'/"Provider").attribute("Name").text rescue nil
```
Comment (repeatedly) [inline lib/fluent/plugin/parser_winevt_xml.rb]: Is call `doc/'Event'/'System'` for each element slow?
How about `system_elem = doc/'Event'/'System'`?

Comment (cosmo0920) [reply lib/fluent/plugin/parser_winevt_xml.rb]: Oops. I've fixed in https://github.com/fluent/fluent-plugin-windows-eventlog/pull/21/commits/2ad9629be9f6c5858fa5265b19643e8b43558fe1.

**Comment by repeatedly** [comment]
If this is ready, we can merge the patch and release new version for users.

**Comment by cosmo0920** [comment]
Dependent gem's memory leak issue is solved.
This PR ready to be merged.

**Comment by repeatedly** [comment]
Awesome work!

**Comment by repeatedly** [comment]
@cosmo0920 I added you to this gem owner. Could you release new version? On my mac, I can't install winevt_c gem so hard to release via `rake release`. I have already pushed version up and changelog.

**Comment by cosmo0920** [comment]
Done!

**Pull request merged**